### PR TITLE
t-on-.prevent.stop.capture.noupdate

### PR DESCRIPTION
**Pull request by bso-odoo**

### CODE REVIEW

File: addons/website/static/tests/core/interaction.test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { beforeEach, describe, expect, test } from "@odoo/hoot";
 import { animationFrame, click, dblclick } from "@odoo/hoot-dom";
 import { advanceTime, Deferred } from "@odoo/hoot-mock";
 import { patchWithCleanup } from "@web/../tests/web_test_helpers";
-import { Colibri } from "@website/core/colibri";
+import { Colibri, SKIP_IMPLICIT_UPDATE } from "@website/core/colibri";
```
Comment (ged-odoo) [inline addons/website/static/tests/core/interaction.test.js]: i don't think this import is necessary...

File: addons/website/static/src/core/colibri.js [hunk 2]
```diff
@@ -47,6 +47,31 @@ export class Colibri {
...
+                    options ||= {};
+                    options.capture = true;
+                    return f;
+                },
+                noupdate: (f) => ((ev) => {
+                    f(ev);
+                    return SKIP_IMPLICIT_UPDATE;
+                }),
+            }[groups.suffix](fn);
+            event = groups.event;
```
Comment (ged-odoo) [inline addons/website/static/src/core/colibri.js:72]:
```js
while (groups) {
    switch (groups.suffix) {
        case "prevent":
            fn = (ev) => {
                ev.preventDefault();
                return fn(ev)
            };
            break;
        case "stop":
            fn = (ev) => {
                ev.stopPropagation();
                return fn(ev)
            };
            break;
        case "capture":
            options ||= {};
            options.capture = true;
            break;
        case "noupdate":
            fn = (ev) => {
                f(ev);
                return SKIP_IMPLICIT_UPDATE;
            };
            break;
        default:
            throw new Error("Invalid t-on- suffix (should be 'prevent', 'stop', 'capture', 'noupdate')");
        }
    }
    event = groups.event;
    groups = re.exec(event)?.groups;
}
```

Comment (bso-odoo) [reply addons/website/static/src/core/colibri.js:72]: This then also requires to distinguish each depth function to avoid a recursive loop:
=>
```js
case ...: {
     let oldFn = fn;
     fn = (ev) => { .... oldFn(ev) ... }
     break;
}
```

TL;DR we keep the lookup object
